pkg/endpoints/endpoints_test.go

```diff
@@ -0,0 +1,162 @@
+				{
+					ID:     "vpc",
+					URL:    "https://vpchost:8080",
+					Region: "eu-gb",
+				},
+			},
+			expectedError: nil,
+		},
+		{
+			name:        "single region, single powervs service",
+			flagToParse: "lon:powervs=https://pvshost:8080",
+			expectedOutput: []ServiceEndpoint{
+				{
+					ID:     "powervs",
+					URL:    "https://pvshost:8080",
+					Region: "lon",
+				},
+			},
+			expectedError: nil,
+		},
+		{
+			name:        "single region, multiple services",
+			flagToParse: "lon:powervs=https://pvshost:8080,rc=https://rchost:8080",
+			expectedOutput: []ServiceEndpoint{
+				{
+					ID:     "powervs",
+					URL:    "https://pvshost:8080",
+					Region: "lon",
+				},
+				{
+					ID:     "rc",
+					URL:    "https://rchost:8080",
+					Region: "lon",
+				},
+			},
+			expectedError: nil,
+		},
+		{
+			name:           "single region, duplicate service",
+			flagToParse:    "eu-gb:vpc=https://localhost:8080,vpc=https://vpchost:8080",
+			expectedOutput: nil,
+			expectedError:  errServiceEndpointDuplicateID,
+		},
+		{
+			name:           "single region, non-valid URI",
+			flagToParse:    "eu-gb:vpc=fdsfs",
+			expectedOutput: nil,
+			expectedError:  errServiceEndpointURL,
+		},
+		{
+			name:        "multiples regions",
+			flagToParse: "eu-gb:vpc=https://vpchost:8080;lon:powervs=https://pvshost:8080,rc=https://rchost:8080",
+			expectedOutput: []ServiceEndpoint{
+				{
+					ID:     "vpc",
+					URL:    "https://vpchost:8080",
+					Region: "eu-gb",
+				},
+				{
+					ID:     "powervs",
+					URL:    "https://pvshost:8080",
+					Region: "lon",
+				},
+				{
+					ID:     "rc",
+					URL:    "https://rchost:8080",
+					Region: "lon",
+				},
+			},
+			expectedError: nil,
+		},
+		{
+			name:        "multiples regions",
+			flagToParse: "eu-gb:vpc=https://vpchost:8080;lon:powervs=https://pvshost:8080,rc=https://rchost:8080;us-south:powervs=https://pvshost-us:8080",
+			expectedOutput: []ServiceEndpoint{
+				{
+					ID:     "vpc",
+					URL:    "https://vpchost:8080",
+					Region: "eu-gb",
+				},
+				{
+					ID:     "powervs",
+					URL:    "https://pvshost:8080",
+					Region: "lon",
+				},
+				{
+					ID:     "rc",
+					URL:    "https://rchost:8080",
+					Region: "lon",
+				},
+				{
+					ID:     "powervs",
+					URL:    "https://pvshost-us:8080",
+					Region: "us-south",
+				},
+			},
+			expectedError: nil,
+		},
+		{
+			name:           "invalid config",
+			flagToParse:    "eu-gb=localhost",
+			expectedOutput: nil,
+			expectedError:  errServiceEndpointRegion,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			out, err := ParseServiceEndpointFlag(tc.flagToParse)
+
+			if !errors.Is(err, tc.expectedError) {
+				t.Fatalf("did not expect correct error: got %v, expected %v", err, tc.expectedError)
+			}
+
+			if !assert.ElementsMatch(t, out, tc.expectedOutput) {
+				t.Fatalf("did not expect correct output: got %v, expected %v", out, tc.expectedOutput)
+			}
```

Comment:
you don't need that if and fatal block, that assert will take care of that
```suggestion
			assert.ElementsMatch(t, out, tc.expectedOutput)
```

Reply:
Fixed.